### PR TITLE
Virtual router crd to sdk conversions

### PR DESCRIPTION
**Pull request by fawadkhaliq**
Added conversion functions for virtual router type 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/conversions/types_conversion_test.go [hunk 1]
```diff
@@ -49,3 +49,44 @@ func TestConvert_CRD_PortMapping_To_SDK_PortMapping(t *testing.T) {
...
+		args       args
+		wantSDKObj *appmeshsdk.Duration
+		wantErr    error
+	}{
+		{
+			name: "normal case",
+			args: args{
+				crdObj: &appmesh.Duration{
+					Unit:  "ms",
+					Value: int64(200),
```
Comment (M00nF1sh) [inline pkg/conversions/types_conversion_test.go:70]: i think this didn't compile.. Unit is a enum and Value is a int64 value instead of pointer in crd

Comment (M00nF1sh) [reply pkg/conversions/types_conversion_test.go:70]: my bad, i thought it's aws.Int64.

File: pkg/conversions/virtualrouter_types_conversion.go [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+	return nil
+}
+
+func Convert_CRD_HeaderMatchMethod_To_SDK_HeaderMatchMethod(crdObj *appmesh.HeaderMatchMethod,
+	sdkObj *appmeshsdk.HeaderMatchMethod, scope conversion.Scope) error {
+
+	sdkObj.Exact = crdObj.Exact
+	sdkObj.Prefix = crdObj.Prefix
+
+	sdkObj.Range = &appmeshsdk.MatchRange{}
```
Comment (M00nF1sh) [inline pkg/conversions/virtualrouter_types_conversion.go:41]: we need to check crdObj == nil here.
If crdObj == nil, sdkObj.Range should be nil. also(these Convert*'s input should be non-nil)

File: pkg/conversions/virtualrouter_types_conversion.go [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+	return nil
+}
+
+func Convert_CRD_HTTPRouteHeader_To_SDK_HttpRouteHeader(crdObj *appmesh.HTTPRouteHeader,
+	sdkObj *appmeshsdk.HttpRouteHeader, scope conversion.Scope) error {
+
+	sdkObj.Name = aws.String(crdObj.Name)
+
+	sdkObj.Match = &appmeshsdk.HeaderMatchMethod{}
+	if err := Convert_CRD_HeaderMatchMethod_To_SDK_HeaderMatchMethod(crdObj.Match, sdkObj.Match, scope); err != nil {
```
Comment (M00nF1sh) [inline pkg/conversions/virtualrouter_types_conversion.go:57]: we need to check crdObj.Match==nil here

File: pkg/conversions/virtualrouter_types_conversion.go [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+	if err := Convert_CRD_HTTPRouteMatch_To_SDK_HttpRouteMatch(&crdObj.Match, sdkObj.Match, scope); err != nil {
+		return err
+	}
+
+	sdkObj.Action = &appmeshsdk.HttpRouteAction{}
+	if err := Convert_CRD_HTTPRouteAction_To_SDK_HttpRouteAction(&crdObj.Action, sdkObj.Action, scope); err != nil {
+		return err
+	}
+
+	sdkObj.RetryPolicy = &appmeshsdk.HttpRetryPolicy{}
```
Comment (M00nF1sh) [inline pkg/conversions/virtualrouter_types_conversion.go:149]: we need to check crdObj. RetryPolicy nil here

File: pkg/conversions/virtualrouter_types_conversion.go [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+}
+
+func Convert_CRD_GRPCRouteMetadataMatchMethod_To_SDK_GrpcRouteMetadataMatchMethod(crdObj *appmesh.GRPCRouteMetadataMatchMethod,
+	sdkObj *appmeshsdk.GrpcRouteMetadataMatchMethod, scope conversion.Scope) error {
+
+	sdkObj.Exact = crdObj.Exact
+	sdkObj.Prefix = crdObj.Prefix
+
+	sdkObj.Range = &appmeshsdk.MatchRange{}
+	if err := Convert_CRD_MatchRange_To_SDK_MatchRange(crdObj.Range, sdkObj.Range, scope); err != nil {
```
Comment (M00nF1sh) [inline pkg/conversions/virtualrouter_types_conversion.go:185]: we need to check crdObj.Range nil here

File: pkg/conversions/virtualrouter_types_conversion.go [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+	sdkObj.Suffix = crdObj.Suffix
+	return nil
+}
+
+func Convert_CRD_GRPCRouteMetadata_To_SDK_GrpcRouteMetadata(crdObj *appmesh.GRPCRouteMetadata,
+	sdkObj *appmeshsdk.GrpcRouteMetadata, scope conversion.Scope) error {
+
+	sdkObj.Name = aws.String(crdObj.Name)
+
+	sdkObj.Match = &appmeshsdk.GrpcRouteMetadataMatchMethod{}
```
Comment (M00nF1sh) [inline pkg/conversions/virtualrouter_types_conversion.go:199]: we need to check crdObj.Match nil here

File: pkg/conversions/virtualrouter_types_conversion.go [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+	if err := Convert_CRD_GRPCRouteMatch_To_SDK_GrpcRouteMatch(&crdObj.Match, sdkObj.Match, scope); err != nil {
+		return err
+	}
+
+	sdkObj.Action = &appmeshsdk.GrpcRouteAction{}
+	if err := Convert_CRD_GRPCRouteAction_To_SDK_GrpcRouteAction(&crdObj.Action, sdkObj.Action, scope); err != nil {
+		return err
+	}
+
+	sdkObj.RetryPolicy = &appmeshsdk.GrpcRetryPolicy{}
```
Comment (M00nF1sh) [inline pkg/conversions/virtualrouter_types_conversion.go:300]: we need to check crdObj.RetryPolicy nil here

File: pkg/conversions/virtualrouter_types_conversion.go [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+	return Convert_CRD_GRPCRetryPolicy_To_SDK_GrpcRetryPolicy(crdObj.RetryPolicy, sdkObj.RetryPolicy, scope)
+}
+
+func Convert_CRD_Route_To_SDK_RouteSpec(crdObj *appmesh.Route, sdkObj *appmeshsdk.RouteSpec, scope conversion.Scope) error {
+
+	sdkObj.GrpcRoute = &appmeshsdk.GrpcRoute{}
```
Comment (M00nF1sh) [inline pkg/conversions/virtualrouter_types_conversion.go:306]: we need to check nil for these routes here

File: pkg/conversions/virtualrouter_types_conversion.go (hunk 2 above)
Comment (fawadkhaliq) [reply pkg/conversions/virtualrouter_types_conversion.go:41]: Ah I totally missed that. Thanks for pointing it out. I'll address here and all the other places

Comment (fawadkhaliq) [reply pkg/conversions/virtualrouter_types_conversion.go]: Done

File: pkg/conversions/virtualrouter_types_conversion.go (hunk 3 above)
Comment (fawadkhaliq) [reply pkg/conversions/virtualrouter_types_conversion.go]: Done

File: pkg/conversions/virtualrouter_types_conversion.go (hunk 4 above)
Comment (fawadkhaliq) [reply pkg/conversions/virtualrouter_types_conversion.go]: Done

File: pkg/conversions/virtualrouter_types_conversion.go (hunk 5 above)
Comment (fawadkhaliq) [reply pkg/conversions/virtualrouter_types_conversion.go]: Done

File: pkg/conversions/virtualrouter_types_conversion.go (hunk 6 above)
Comment (fawadkhaliq) [reply pkg/conversions/virtualrouter_types_conversion.go]: Done

File: pkg/conversions/virtualrouter_types_conversion.go (hunk 7 above)
Comment (fawadkhaliq) [reply pkg/conversions/virtualrouter_types_conversion.go]: Done

File: pkg/conversions/virtualrouter_types_conversion.go (hunk 8 above)
Comment (fawadkhaliq) [reply pkg/conversions/virtualrouter_types_conversion.go]: Done. Thanks!

**Pull request merged**